### ISSUE TITLE
Please elaborate on the decision to not use GDscript

### ISSUE THREAD

**Issue by unfa** [sboron/godot4-fast-paced-network-fps-tps#4]
The project readme states:

> Totaly written in c#. No gdscript, because its sucks hard for oop.

I assume oop means Object-Oriented Programming here.
I'd love to know what issues have you encountered.

I am a lead of a [multiplayer FPS game in Godot 4](https://libla.st) project, though we use GDscript.
I'd love to use this project for reference and borrow ideas (though C# is gonna make that more difficult for me).

**Comment by jamesdimick** [comment]
I for one like that it is in C#. I prefer the language to GDScript. I've never really liked GDScript, or Python for that matter. It seems more like a language you use when you're first learning to program, then later you "upgrade" to a more strict language like C#, C++, etc.

If you have the proper programming skills, converting things between languages isn't all that hard. I've converted many things from GDScript to C#, for example.

But, to each their own...

**Comment by sboron** [comment]
Golden Rule. the bigger the project, the stricter the tech stack should be.

**Comment by sboron** [comment]
Golden Rule. the bigger the project, the stricter the tech stack should be.

**Comment by unfa** [comment]
Static typing in GDscript2 (Godot 4) is very much an option and doing that everywhere allows for better performance (not to mention easier debugging).

I understand that stricter languages force good practices better, but I personally value the stripped down syntax and enhanced readability of GDscript.

I'll keep an eye out and hope to learn from your project (especially in regards to implementing proper networking).

**Comment by sboron** [comment]
First of all, C# is quite a bit faster than GDScript (when used properly). Secondly, such a complex system is not suitable for prototyping languages like GDScript (yes GDScript is purely for prototyping, fast to finish but messy when it becomes a big project). But there are also other reasons such as mathematical precision, better thread action, etc.

**Comment by sboron** [comment]
Of course, you could also rewrite the project in GDScript, but that actually makes little sense, since the performance then also drops. A possibility would be however GDNative.

**Comment by sboron** [comment]
Incidentally, please post such topics under "Discussions" and not under "Issues".
